ambry-store/src/main/java/com/github/ambry/store/HybridCompactionPolicy.java

```diff
@@ -0,0 +1,191 @@
+    backUpCompactionPolicyInfo(dataDir, compactionPolicySwitchInfo);
+    compactionPolicySwitchInfo.getCompactionPolicyCounter().increment();
+    return selectCompactionPolicy.getCompactionDetails(totalCapacity, usedCapacity, segmentCapacity, segmentHeaderSize,
+        logSegmentsNotInJournal, blobStoreStats, dataDir);
+  }
+
+  /**
+   * Get the CompactionPolicySwitchInfo from file or blobToCompactionPolicySwitchInfoMap.
+   * @param dataDir The directory to store the file.
+   * @param storeId id of the BlobStore
+   * @return {@link CompactionPolicySwitchInfo} gets from map or recover from file if needed.
+   */
+  private CompactionPolicySwitchInfo getCompactionPolicySwitchInfo(String storeId, String dataDir, BlobStoreStats blobStoreStats) {
+    if (!blobToCompactionPolicySwitchInfoMap.containsKey(storeId)) {
+      File file = new File(Paths.get(dataDir, COMPACT_POLICY_INFO_PATH).toString());
+      if (file.exists()) {
+        CompactionPolicySwitchInfo compactionPolicySwitchInfo = recoverCompactionPolicySwitchInfo(file, blobStoreStats);
+        blobToCompactionPolicySwitchInfoMap.put(storeId, compactionPolicySwitchInfo);
+      } else {
+        blobToCompactionPolicySwitchInfoMap.put(storeId,
+            new CompactionPolicySwitchInfo(new CompactionPolicyCounter(storeConfig.storeCompactionPolicySwitchCounterDays),
+                INIT_COMPACT_ALL_TIME));
+      }
+    }
+    return blobToCompactionPolicySwitchInfoMap.get(storeId);
+  }
+
+  /**
+   * Recover the {@link CompactionPolicySwitchInfo} from backup file.
+   * {
+   *   "compactionPolicyCounter" : {
+   *     "storeCompactionPolicySwitchCounterDays" : 3,
+   *     "counter" : 1
+   *   },
+   *   "lastCompactAllTime" : 1593492962651
+   * }
+   * @param file the backup file stores {@link CompactionPolicySwitchInfo}
+   */
+  private CompactionPolicySwitchInfo recoverCompactionPolicySwitchInfo(File file, BlobStoreStats blobStoreStats) {
+    try {
+      return objectMapper.readValue(file, CompactionPolicySwitchInfo.class);
+    } catch (IOException e) {
+      logger.error("Could not deserialize file : {} into {} Object", file, CompactionPolicySwitchInfo.class.getName());
+      blobStoreStats.getMetrics().blobStoreRecoverCompactionPolicySwitchInfoErrorCount.inc();
+      return new CompactionPolicySwitchInfo(new CompactionPolicyCounter(storeConfig.storeCompactionPolicySwitchCounterDays),
+          INIT_COMPACT_ALL_TIME);
+    }
+  }
+
+  /**
+   * Selects which compaction policy to use for current compaction cycle.
+   * @return {@link CompactionPolicy} to use for current compaction cycle(Selects between {@link StatsBasedCompactionPolicy} and {@link CompactAllPolicy}).
+   */
+  CompactionPolicy selectCompactionPolicyAndUpdateCompactionPolicySwitchInfo(
+      CompactionPolicySwitchInfo compactionPolicySwitchInfo) {
+    if (compactionPolicySwitchInfo == null) {
+      logger.trace("CompactionPolicySwitchInfo is null");
```

Comment:
nit: please add datadir or store id in the log.

Reply:
Done.